subs/pantry/src/Pantry/Storage.hs

```diff
@@ -243,39 +258,61 @@ withStorage
 withStorage action =
   flip SQLite.withStorage_ action =<< view (P.pantryConfigL.to P.pcStorage)
 
+
+rdbmsAwareQuery
```

Comment:
Please add some docs to explain what the different arguments are here. I'm tempted to say there should be a datatype helper to avoid positional arguments, e.g.:

```haskell
data RdmbsActions m a = RdbmsActions
   { raSqlite :: !(ReaderT SqlBackend m a)
   , raPostgres :: !(ReaderT SqlBackend m a)
   }
```

Reply:
This is only an internal function that is used in two places, an extra data type seems like an overkill. But if you feel strongly about it, I can add `RdmbsActions`.

I will add some documentation explaining this function.

Reply:
I don't think it's overkill at all, we could very easily write incorrect code without the types there to help us.

Reply:
Done. Added `RdbmsActions` data type and some documentation.